diff --git a/Makefile b/Makefile
index 875754a6b8..e1f8dfb27a 100644
--- a/Makefile
+++ b/Makefile
@@ -150,7 +150,7 @@ GOINSTALL=$(GOCMD) install
 GOTEST=$(GOCMD) test
 GODEP=$(GOTEST) -i
 GOFMT=gofmt -w
-GOBUILDIMAGE=golang:1.13.8
+GOBUILDIMAGE=arm64v8/golang:1.13.8
 GOBUILDPATHINCONTAINER=/harbor

 # go build
@@ -298,7 +298,7 @@ SWAGGER_IMAGENAME=goharbor/swagger
 SWAGGER_VERSION=v0.21.0
 SWAGGER=$(DOCKERCMD) run --rm -u $(shell id -u):$(shell id -g) -v $(BUILDPATH):$(BUILDPATH) -w $(BUILDPATH) ${SWAGGER_IMAGENAME}:${SWAGGER_VERSION}
 SWAGGER_GENERATE_SERVER=${SWAGGER} generate server --template-dir=$(TOOLSPATH)/swagger/templates --exclude-main
-SWAAGER_IMAGE_BUILD_CMD=${DOCKERBUILD} -f ${TOOLSPATH}/swagger/Dockerfile --build-arg SWAGGER_VERSION=${SWAGGER_VERSION} -t ${SWAGGER_IMAGENAME}:$(SWAGGER_VERSION) .
+SWAAGER_IMAGE_BUILD_CMD=${DOCKERBUILD} -f ${TOOLSPATH}/swagger/Dockerfile --build-arg ARCH=${ARCH} --build-arg SWAGGER_VERSION=${SWAGGER_VERSION} -t ${SWAGGER_IMAGENAME}:$(SWAGGER_VERSION) .
 
 SWAGGER_IMAGENAME:
 	@if [ "$(shell ${DOCKERIMASES} -q ${SWAGGER_IMAGENAME}:$(SWAGGER_VERSION) 2> /dev/null)" == "" ]; then \
@@ -383,8 +383,7 @@ build:
 build_base_docker:
        @for name in chartserver clair clair-adapter trivy-adapter core db jobservice log nginx notary-server notary-signer portal prepare redis registry registryctl; do \
                echo $$name ; \
-               $(DOCKERBUILD) --pull -f $(MAKEFILEPATH_PHOTON)/$$name/Dockerfile.base -t $(BASEIMAGENAMESPACE)/harbor-$$name-base:$(BASEIMAGETAG) . && \
-               $(PUSHSCRIPTPATH)/$(PUSHSCRIPTNAME) $(BASEIMAGENAMESPACE)/harbor-$$name-base:$(BASEIMAGETAG) $(REGISTRYUSER) $(REGISTRYPASSWORD) || exit 1; \
+               $(DOCKERBUILD) --pull -f $(MAKEFILEPATH_PHOTON)/$$name/Dockerfile.base -t $(BASEIMAGENAMESPACE)/harbor-$$name-base:$(BASEIMAGETAG) .; \
        done

 pull_base_docker:
diff --git a/make/photon/chartserver/Dockerfile.base b/make/photon/chartserver/Dockerfile.base
index ecf7b90343..a60344de9e 100644
--- a/make/photon/chartserver/Dockerfile.base
+++ b/make/photon/chartserver/Dockerfile.base
@@ -1,4 +1,5 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install -y shadow sudo >>/dev/null\
     && tdnf clean all \
diff --git a/make/photon/clair-adapter/Dockerfile.base b/make/photon/clair-adapter/Dockerfile.base
index 80b257f60e..38bf61cfe9 100644
--- a/make/photon/clair-adapter/Dockerfile.base
+++ b/make/photon/clair-adapter/Dockerfile.base
@@ -1,7 +1,8 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install -y sudo >>/dev/null\
     && tdnf clean all \
     && mkdir /clair-adapter/ \
     && groupadd -r -g 10000 clair-adapter \
-    && useradd --no-log-init -m -r -g 10000 -u 10000 clair-adapter
\ No newline at end of file
+    && useradd --no-log-init -m -r -g 10000 -u 10000 clair-adapter
diff --git a/make/photon/clair/Dockerfile.base b/make/photon/clair/Dockerfile.base
index 8f9737450e..3deab74d1b 100644
--- a/make/photon/clair/Dockerfile.base
+++ b/make/photon/clair/Dockerfile.base
@@ -1,6 +1,7 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install -y git shadow sudo rpm xz python-xml >>/dev/null\
     && tdnf clean all \
     && groupadd -r -g 10000 clair \
-    && useradd --no-log-init -m -g 10000 -u 10000 clair
\ No newline at end of file
+    && useradd --no-log-init -m -g 10000 -u 10000 clair
diff --git a/make/photon/core/Dockerfile.base b/make/photon/core/Dockerfile.base
index 865834fad9..32b6e8e190 100644
--- a/make/photon/core/Dockerfile.base
+++ b/make/photon/core/Dockerfile.base
@@ -1,4 +1,5 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install sudo tzdata -y >> /dev/null \
     && tdnf clean all \
diff --git a/make/photon/db/Dockerfile.base b/make/photon/db/Dockerfile.base
index 3f8e6fdb6a..ca9fda8ad1 100644
--- a/make/photon/db/Dockerfile.base
+++ b/make/photon/db/Dockerfile.base
@@ -1,4 +1,5 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 ENV PGDATA /var/lib/postgresql/data
 
diff --git a/make/photon/jobservice/Dockerfile.base b/make/photon/jobservice/Dockerfile.base
index 5bbd681380..c3820e7403 100644
--- a/make/photon/jobservice/Dockerfile.base
+++ b/make/photon/jobservice/Dockerfile.base
@@ -1,4 +1,5 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install sudo tzdata -y >> /dev/null \
     && tdnf clean all \
diff --git a/make/photon/log/Dockerfile.base b/make/photon/log/Dockerfile.base
index b724481309..c359758709 100644
--- a/make/photon/log/Dockerfile.base
+++ b/make/photon/log/Dockerfile.base
@@ -1,4 +1,5 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install -y cronie rsyslog logrotate shadow tar gzip sudo >> /dev/null\
     && mkdir /var/spool/rsyslog \
diff --git a/make/photon/nginx/Dockerfile.base b/make/photon/nginx/Dockerfile.base
index 8982dc00e1..e2d8915d8e 100644
--- a/make/photon/nginx/Dockerfile.base
+++ b/make/photon/nginx/Dockerfile.base
@@ -1,4 +1,5 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install sudo nginx -y >> /dev/null\
     && tdnf clean all \
diff --git a/make/photon/notary-server/Dockerfile.base b/make/photon/notary-server/Dockerfile.base
index da4b03cbe2..99c940379f 100644
--- a/make/photon/notary-server/Dockerfile.base
+++ b/make/photon/notary-server/Dockerfile.base
@@ -1,4 +1,5 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install -y shadow sudo \
     && tdnf clean all \
diff --git a/make/photon/notary-signer/Dockerfile.base b/make/photon/notary-signer/Dockerfile.base
index da4b03cbe2..99c940379f 100644
--- a/make/photon/notary-signer/Dockerfile.base
+++ b/make/photon/notary-signer/Dockerfile.base
@@ -1,4 +1,5 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install -y shadow sudo \
     && tdnf clean all \
diff --git a/make/photon/prepare/Dockerfile.base b/make/photon/prepare/Dockerfile.base
index c8a1f6bf52..7a823a82c8 100644
--- a/make/photon/prepare/Dockerfile.base
+++ b/make/photon/prepare/Dockerfile.base
@@ -1,4 +1,5 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
-RUN tdnf install -y python3 python3-pip httpd && tdnf clean all
+RUN tdnf install -y python3 python3-pip httpd
 RUN pip3 install setuptools && pip3 install pipenv==2018.11.26
diff --git a/make/photon/redis/Dockerfile.base b/make/photon/redis/Dockerfile.base
index d800f6a914..a26ee41b50 100644
--- a/make/photon/redis/Dockerfile.base
+++ b/make/photon/redis/Dockerfile.base
@@ -1,3 +1,4 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install -y redis sudo && tdnf clean all
diff --git a/make/photon/registry/Dockerfile.base b/make/photon/registry/Dockerfile.base
index 0e618e1b8c..e9fe738b2c 100644
--- a/make/photon/registry/Dockerfile.base
+++ b/make/photon/registry/Dockerfile.base
@@ -1,6 +1,7 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install sudo -y >> /dev/null\
     && tdnf clean all \
     && mkdir -p /etc/registry \
-    && groupadd -r -g 10000 harbor && useradd --no-log-init -m -g 10000 -u 10000 harbor
\ No newline at end of file
+    && groupadd -r -g 10000 harbor && useradd --no-log-init -m -g 10000 -u 10000 harbor
diff --git a/make/photon/registryctl/Dockerfile.base b/make/photon/registryctl/Dockerfile.base
index c536863410..da294d0a80 100644
--- a/make/photon/registryctl/Dockerfile.base
+++ b/make/photon/registryctl/Dockerfile.base
@@ -1,4 +1,5 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install sudo -y >> /dev/null \
     && tdnf clean all \
diff --git a/make/photon/trivy-adapter/Dockerfile.base b/make/photon/trivy-adapter/Dockerfile.base
index 3b38817dce..4570138576 100644
--- a/make/photon/trivy-adapter/Dockerfile.base
+++ b/make/photon/trivy-adapter/Dockerfile.base
@@ -1,4 +1,5 @@
-FROM photon:2.0
+FROM arm64v8/photon:3.0
+COPY qemu-aarch64-static /usr/bin
 
 RUN tdnf install -y sudo rpm >> /dev/null \
     && tdnf clean all \
diff --git a/tools/swagger/Dockerfile b/tools/swagger/Dockerfile
index 74e5fffab3..c9fce328f2 100644
--- a/tools/swagger/Dockerfile
+++ b/tools/swagger/Dockerfile
@@ -1,7 +1,8 @@
 FROM golang:1.13.8
 
 ARG SWAGGER_VERSION
-RUN curl -fsSL -o /usr/bin/swagger https://github.com/go-swagger/go-swagger/releases/download/$SWAGGER_VERSION/swagger_linux_amd64 && chmod +x /usr/bin/swagger
+ARG ARCH
+RUN curl -fsSL -o /usr/bin/swagger https://github.com/go-swagger/go-swagger/releases/download/$SWAGGER_VERSION/swagger_linux_$ARCH && chmod +x /usr/bin/swagger
 
 ENTRYPOINT ["/usr/bin/swagger"]
 CMD ["--help"]
